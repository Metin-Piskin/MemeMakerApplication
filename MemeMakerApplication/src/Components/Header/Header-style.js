import { Dimensions, StyleSheet } from "react-native";

export default StyleSheet.create({
    homecontainer: {
        backgroundColor: '#161718',
        //borderWidth: 1,
        borderBottomWidth: 1,
        borderColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
        marginHorizontal: 5,
        paddingVertical: 5
    },
    hometext: {
        fontFamily: 'SpecialElite-Regular',
        fontSize: 30,
        color: '#fff'
    },
    homepluscontainer: {
        // borderWidth: 1,
        marginLeft: 45,
        borderRadius: 5,
        borderColor: '#fff'
    },
    homeplus: {
        // padding: 5,
        color: '#fff',
        fontSize: 44,
        paddingHorizontal: 10,
    },
    detailcontainer: {
        backgroundColor: '#161718',
        borderColor: '#fff',
        borderBottomWidth: 1,
        flexDirection: 'row',
        marginHorizontal: 5,
        paddingVertical: 5,
    },
    backbutton: {
        alignSelf: 'center',
    },
    text: {
        alignSelf: 'center',
        fontSize: 30,
        fontFamily: 'SpecialElite-Regular',
        color: '#fff',
        justifyContent: 'center',
        left: Dimensions.get('screen').width / 2.95
    },

})