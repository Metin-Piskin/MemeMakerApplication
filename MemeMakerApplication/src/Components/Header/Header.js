import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import Ionicons from 'react-native-vector-icons/Ionicons'

import styles from './Header-style';

const Header = ({ home, backPress }) => {
    return (
        home ? (
            <View style={styles.homecontainer} testID='header-home-container'>
                <Text style={styles.hometext} testID='header-home-home-text'>
                    Meme Maker
                </Text>
            </View>
        ) : (
            <View style={styles.detailcontainer}  testID='header-home-detail-container'>
                <TouchableOpacity
                    style={styles.backbutton}
                    testID='header-home-back-button'
                    onPress={backPress}
                >
                    <Ionicons
                        name='caret-back'
                        size={35}
                        color='#fff'
                    />
                </TouchableOpacity>
                <Text style={styles.text}  testID='header-home-text'>
                    Edit
                </Text>
            </View>
        )

    )
}

export default Header;