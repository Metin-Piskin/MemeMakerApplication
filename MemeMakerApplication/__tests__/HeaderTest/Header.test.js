import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import Header from '../../src/Components/Header';
import HeaderStyles from '../../src/Components/Header/Header-style';

jest.mock('react-native-vector-icons/Ionicons', () => 'Ionicons')

test('renders correctly', () => {
    const test = true
    const comp = render(<Header home={test} />);
    expect(comp).toMatchSnapshot();
});

test('should render default theme style', () => {
    const test = true
    const comp = render(<Header home={test} />);
    const HeaderHomeContainerStyle = comp.getByTestId('header-home-container').props.style;

    expect(HeaderHomeContainerStyle).toMatchObject(HeaderStyles.homecontainer);
});

test('should render default theme style', () => {
    const test = true
    const comp = render(<Header home={test} />);
    const HeaderHomeTextStyle = comp.getByTestId('header-home-home-text').props.style;

    expect(HeaderHomeTextStyle).toMatchObject(HeaderStyles.hometext);
});

test('renders correctly', () => {
    const comp = render(<Header />);
    expect(comp).toMatchSnapshot();
});

test('should render press correctly', () => {
    const mockFunction = jest.fn();
    const comp = render(<Header backPress={mockFunction} />);

    const HeaderBackButtonPress = comp.getByTestId('header-home-back-button');
    fireEvent(HeaderBackButtonPress, 'press');

    expect(mockFunction).toBeCalled();
})

test('should render default theme style', () => {
    const comp = render(<Header />);
    const HeaderHomeDetailContainerStyle = comp.getByTestId('header-home-detail-container').props.style;

    expect(HeaderHomeDetailContainerStyle).toMatchObject(HeaderStyles.detailcontainer);
});

test('should render default theme style', () => {
    const comp = render(<Header />);
    const HeaderHomeBackButtonStyle = comp.getByTestId('header-home-back-button').props.style;

    expect(HeaderHomeBackButtonStyle).toMatchObject(HeaderStyles.backbutton);
});

test('should render default theme style', () => {
    const comp = render(<Header />);
    const HeaderTextStyle = comp.getByTestId('header-home-text').props.style;

    expect(HeaderTextStyle).toMatchObject(HeaderStyles.text);
});
