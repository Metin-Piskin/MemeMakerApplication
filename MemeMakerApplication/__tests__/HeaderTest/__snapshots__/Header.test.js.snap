// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#161718",
      "borderBottomWidth": 1,
      "borderColor": "#fff",
      "justifyContent": "center",
      "marginHorizontal": 5,
      "paddingVertical": 5,
    }
  }
  testID="header-home-container"
>
  <Text
    style={
      {
        "color": "#fff",
        "fontFamily": "SpecialElite-Regular",
        "fontSize": 30,
      }
    }
    testID="header-home-home-text"
  >
    Meme Maker
  </Text>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    {
      "backgroundColor": "#161718",
      "borderBottomWidth": 1,
      "borderColor": "#fff",
      "flexDirection": "row",
      "marginHorizontal": 5,
      "paddingVertical": 5,
    }
  }
  testID="header-home-detail-container"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "center",
        "opacity": 1,
      }
    }
    testID="header-home-back-button"
  >
    <Ionicons
      color="#fff"
      name="caret-back"
      size={35}
    />
  </View>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#fff",
        "fontFamily": "SpecialElite-Regular",
        "fontSize": 30,
        "justifyContent": "center",
        "left": 254.23728813559322,
      }
    }
    testID="header-home-text"
  >
    Edit
  </Text>
</View>
`;
